src/mongocxx/test/spec/change_stream.cpp

```diff
@@ -41,79 +35,96 @@ using namespace bsoncxx;
-        if (test["description"].get_utf8().value.compare(
-                "Change Stream should error when an invalid aggregation stage is passed in") == 0) {
-            WARN("Skipping test with invalid pipeline stages. The C++ driver cannot test them.");
-            continue;
-        }
-
-        options::change_stream cs_opts{};
-        if (test["changeStreamOptions"]) {
-            auto options = test["changeStreamOptions"].get_document().value;
-            if (options["batchSize"]) {
-                cs_opts.batch_size(options["batchSize"].get_int32().value);
+        auto description = test["description"].get_utf8().value;
+        SECTION(to_string(description)) {
+            if (should_skip(test_el))
+                continue;
+
+            // "Use globalClient to [drop and recreate the collections]".
+            test_setup(test, test_spec);
+
+            // "Create a new MongoClient `client`"
+            spec::apm_checker apm_checker;
+            options::client client_opts;
+            // "Begin monitoring all APM events for `client`"
+            apm_checker.skip_kill_cursors();
+            client_opts.apm_opts(apm_checker.get_apm_opts(true));
+            class client client(uri{}, client_opts);
+
+            options::change_stream cs_opts{};
+            if (test["changeStreamOptions"]) {
+                auto options = test["changeStreamOptions"].get_document().value;
+                if (options["batchSize"]) {
+                    cs_opts.batch_size(options["batchSize"].get_int32().value);
+                }
             }
-        }
 
-        // "Using client, create a changeStream changeStream against the specified target"
-        pipeline pipeline{};
-        auto cs = [&]() {
-            if (test["changeStreamPipeline"]) {
-                pipeline = build_pipeline(test["changeStreamPipeline"].get_array().value);
-            }
-            auto target = std::string(test["target"].get_utf8().value);
-            if (target == "collection") {
-                return client[db1_name][coll1_name].watch(pipeline, cs_opts);
-            } else if (target == "database") {
-                return client[db1_name].watch(pipeline, cs_opts);
-            } else {
-                return client.watch(pipeline);
-            }
-        }();
+            // "Using client, create a changeStream changeStream against the specified target"
+            auto cs = [&]() {
+                pipeline pipeline{};
+                if (test["changeStreamPipeline"]) {
+                    pipeline = build_pipeline(test["changeStreamPipeline"].get_array().value);
+                }
 
-        // "Using `globalClient`, run every operation in operations in serial against the server."
-        if (test["operations"]) {
+                std::string db_name = to_string(test_spec["database_name"].get_utf8().value);
+                std::string coll_name = to_string(test_spec["collection_name"].get_utf8().value);
```

Comment:
I just moved these names into this lambda since it was the only place they were used. The rest of the additions is from indentation.